physics/lsm_ruc.F90

```diff
@@ -1018,27 +1033,79 @@ subroutine lsm_ruc_run                                            & ! inputs
         endif
 
         chs_lnd (i,j)   = ch_lnd(i) * wind(i) ! compute conductance
-        flhc_lnd(i,j)   = chs_lnd(i,j) * rho(i) * con_cp ! * (1. + 0.84*q2(i,1,j))
+        flhc_lnd(i,j)   = chs_lnd(i,j) * rho(i) * con_cp * (1.+0.84*q2(i,1,j))
         flqc_lnd(i,j)   = chs_lnd(i,j) * rho(i) * wet(i,j)
+
         ! for output
         cmm_lnd(i) = cm_lnd(i) * wind(i)
         chh_lnd(i) = chs_lnd(i,j) * rho(i)
         !
-        snowh_lnd(i,j) = snwdph_lnd(i) * 0.001         ! convert from mm to m
-        sneqv_lnd(i,j) = weasd_lnd(i)                  ! [mm]
-        snfallac_lnd(i,j) = snowfallac_lnd(i)
-        !> -- sanity checks on sneqv and snowh
-        if (sneqv_lnd(i,j) /= 0.0 .and. snowh_lnd(i,j) == 0.0) then
-          snowh_lnd(i,j) = 0.003 * sneqv_lnd(i,j) ! snow density ~300 kg m-3
+        sneqv_lnd(i,j) = weasd_lnd(i)
+        snowh_lnd(i,j) = snwdph_lnd(i) * 0.001 ! convert from mm to m
+
+        if(kdt > 1) then
+        !-- run-total accumulation
+          snfallac_lnd(i,j) = snowfallac_lnd(i)
+          acsn_lnd(i,j)     = acsnow_lnd(i)
+          snomlt_lnd(i,j)   = snowmt_lnd(i)
         endif
 
-        if (snowh_lnd(i,j) /= 0.0 .and. sneqv_lnd(i,j) == 0.0) then
-          sneqv_lnd(i,j) = 300. * snowh_lnd(i,j) ! snow density ~300 kg m-3
-        endif
-
-        if (sneqv_lnd(i,j) > 0. .and. snowh_lnd(i,j) > 0.) then
-          if(sneqv_lnd(i,j)/snowh_lnd(i,j) > 950.) then
-            sneqv_lnd(i,j) = 300. * snowh_lnd(i,j)
+        !> -- sanity checks on sneqv and snowh
+        if (sneqv_lnd(i,j) /= 0.0d0 .and. snowh_lnd(i,j) == 0.0d0) then
```

Comment:
If these need (?) to be doubles, I would suggest importing kind_dbl_prec, alongside kind_phys, from machine.F90.

Reply:
kind_dbl_prec is used here instead of d0.